### PR TITLE
Test-Constructor Injection

### PR DESCRIPTION
**Pull request by yuriyfomin17**
1. Added some tests
2. Some comments refactor
3. Added injection via constructor

### CODE REVIEW

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 1]
```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
...
+
+    /**
+     * <h3>Injects Beans via {@link Autowired} annotation</h3>
+     *
+     * @param beanClass    {@link Class}
+     * @param beanInstance {@link Object}
+     */
+    private <T> void injectBeanViaAutowiredAnnotation(Class<?> beanClass, T beanInstance) throws IllegalAccessException {
+        for (Field field : beanClass.getDeclaredFields()) {
+            field.setAccessible(true);
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java:99]: We should place this field into if block. It is not necessary operation if condition false.

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 2]
```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
...
 
     /**
-     * Scanning a package for classes annotated by {@link Component}.
-     * It creates an instances of such classes and put them into context.
+     * <h3>Scanning a package for classes annotated by {@link Component}.</h3>
+     * <p>It creates an instances of such classes and put them into context.</p>
      */
     private void initiateContext() {
         Reflections reflections = new Reflections(packagePath, Scanners.TypesAnnotated);
+        Set<Class<?>> futureComponents = reflections.getTypesAnnotatedWith(Component.class);
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java]: `futureComponents `name does not describe real name of type of objects with which we work. Probably better simplify name to `classes`

**Review by Mikhail-Mikhailenko — COMMENTED** [review]
Better to create two separate PR for the two tickets on the trello. It will be easier to track the tasks

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 3]
```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
...
+    private <T> void injectBeanViaConstructor(Class<?> beanClass, T beanInstance) throws NoSuchFieldException, IllegalAccessException {
+        Map<Class<?>, String> fieldTypeStringNameMap = new HashMap<>();
+        for (Field field : PrinterServiceDemo.class.getDeclaredFields()){
+            fieldTypeStringNameMap.put(field.getType(), field.getName());
+        }
+        Constructor<?>[] constructors = beanClass.getConstructors();
+        for (Constructor<?> constructor : constructors){
+            for (Parameter parameter: constructor.getParameters()){
+                Class<?> parameterType  = parameter.getType();
+                var autowiredBeansInstance = getBean(parameterType);
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java:122]: Very complicated name `autowiredBeansInstance` we call getBean method and want to get bean. I think `var bean`  will be sufficient

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 4]
```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
...
+    }
+
+    /**
+     * <h3>Resolves bean name</h3>
+     *
+     * @param beanClass {@link Class}
+     * @return {@link String}
+     */
+    private String resolveBeanName(Class<?> beanClass) {
+        String beanCustomNae = beanClass.getAnnotation(Component.class).value();
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java:139]: typo `beanCustomNae`

Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java:139]: typo `beanCustomNae`

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 5]
```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
...
+        if (!beanCustomNae.isEmpty()) return beanCustomNae;
+        return WordUtils.uncapitalize(beanClass.getSimpleName());
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java:141]: We can simplify  this construction : `StringUtils.defaultIfEmpty(beanCustomName,WordUtils.uncapitalize(bean.getSimpleName()));`

**Review by syevtushok — APPROVED** [review]

**Pull request merged**